tachyon/math/polynomials/univariate/univariate_evaluation_domain.h

```diff
@@ -138,10 +139,32 @@ class UnivariateEvaluationDomain : public EvaluationDomain<F, MaxDegree> {
 
   // Compute a FFT.
   [[nodiscard]] constexpr Evals FFT(const DensePoly& poly) const {
+    Evals evals;
+    if (!poly.IsZero()) {
+      evals.evaluations_ = poly.coefficients_.coefficients_;
+#if TACHYON_CUDA
+      // TODO(chokobole): Remove this condition.
+      if constexpr (std::is_same_v<F, bn254::Fr>) {
+        if (icicle_) {
+          evals.evaluations_.resize(size_);
+          CHECK(
+              (*icicle_)->FFT(FFTAlgorithmToIcicleNTTAlgorithm(GetAlgorithm()),
+                              offset_big_int_, evals));
+          return evals;
+        }
+      }
+#endif
+      DoFFT(evals);
+    }
+    return evals;
+  }
+
+  // Compute a FFT.
+  [[nodiscard]] constexpr Evals FFT(DensePoly&& poly) const {
     if (poly.IsZero()) return {};
 
     Evals evals;
```

Comment:
It looks like optimization is possible here too, is that correct?

```suggestion
    Evals evals;
    if (poly.IsZero()) return evals;
```